servers/rendering/rendering_device.cpp

```diff
@@ -164,7 +164,7 @@ void RenderingDevice::_free_dependencies(RID p_id) {
 
 	// Direct dependencies must be freed.
 
-	HashMap<RID, HashSet<RID>>::Iterator E = dependency_map.find(p_id);
+	AHashMap<RID, HashSet<RID>>::Iterator E = dependency_map.find(p_id);
```

Comment:
```suggestion
	AHashMap<RID, HashSet<RID>>::Iterator E = dependency_map.find(p_id);
	if (E) {
		if (E->value.size() == 1) {
			free(*E->value.begin());
		} else {
			while (true) {
				E = dependency_map.find(p_id); // AHashMap doesn`t have stable iterator.
				if (E->value.size() == 0) {
					break;
				}
				free(*E->value.begin());
			}
		}

		dependency_map.erase(p_id);
	}

	// Reverse dependencies must be unreferenced.
	E = reverse_dependency_map.find(p_id);
```
This will fix the crash at the end.